server/utils/forms/inputs/twelveHourClockTimeInput.test.ts

```diff
@@ -0,0 +1,200 @@
+      })
+
+      it('returns a ClockTime value for a PM time', async () => {
+        const request = TestUtils.createRequest({
+          'alarm-time-hour': '1',
+          'alarm-time-minute': '05',
+          'alarm-time-part-of-day': 'pm',
+        })
+
+        const result = await new TwelveHourClockTimeInput(request, 'alarm-time', messages).validate()
+
+        expect(result.value).toEqual(ClockTime.fromTwentyFourHourComponents(13, 5, 0))
+      })
+    })
+
+    it('returns an error when a field is empty', async () => {
+      const request = TestUtils.createRequest({
+        'alarm-time-hour': '',
+        'alarm-time-minute': '09',
+        'alarm-time-part-of-day': 'pm',
+      })
+
+      const result = await new TwelveHourClockTimeInput(request, 'alarm-time', messages).validate()
+
+      expect(result.error).toEqual({
+        errors: [
+          {
+            errorSummaryLinkedField: 'alarm-time-hour',
+            formFields: ['alarm-time-hour'],
+            message: 'The alarm time must include an hour',
+          },
+        ],
+      })
+    })
+
+    it('returns an error when multiple fields are empty', async () => {
+      const request = TestUtils.createRequest({
+        'alarm-time-hour': '',
+        'alarm-time-minute': '',
+        'alarm-time-part-of-day': 'pm',
+      })
+
+      const result = await new TwelveHourClockTimeInput(request, 'alarm-time', messages).validate()
+
+      expect(result.error).toEqual({
+        errors: [
+          {
+            errorSummaryLinkedField: 'alarm-time-hour',
+            formFields: ['alarm-time-hour'],
+            message: 'The alarm time must include an hour',
+          },
+        ],
+      })
+    })
+
+    it('returns an error when part of day is blank', async () => {
+      const request = TestUtils.createRequest({
+        'alarm-time-hour': '1',
+        'alarm-time-minute': '09',
+        'alarm-time-part-of-day': '',
+      })
+
+      const result = await new TwelveHourClockTimeInput(request, 'alarm-time', messages).validate()
+
+      expect(result.error).toEqual({
+        errors: [
+          {
+            errorSummaryLinkedField: 'alarm-time-part-of-day',
+            formFields: ['alarm-time-part-of-day'],
+            message: 'Select whether the alarm time is AM or PM',
+          },
+        ],
+      })
+    })
+
+    it('returns an error when part of day is neither “am” nor “pm”', async () => {
+      const request = TestUtils.createRequest({
+        'alarm-time-hour': '1',
+        'alarm-time-minute': '09',
+        'alarm-time-part-of-day': 'rabbit',
+      })
+
+      const result = await new TwelveHourClockTimeInput(request, 'alarm-time', messages).validate()
+
+      expect(result.error).toEqual({
+        errors: [
+          {
+            errorSummaryLinkedField: 'alarm-time-part-of-day',
+            formFields: ['alarm-time-part-of-day'],
+            message: 'The alarm time must be a real time',
```

Comment:
i think if the user entered 'rabbit' in the AM/PM input, this message isn't going to help much lol

Reply:
Yeah, it's a pretty hypothetical error anyway, since in reality it's a select box in the UI.